### DISCUSSION TITLE
How to deploy apps to html-repo-host service?

### DISCUSSION THREAD

**Discussion by codeyogi911** [SAP/terraform-provider-cloudfoundry#132] [Q&A]
I am working on a project and want to deploy everything using TF. I was able to make other things work. How to deploy app content to repo host?

**Comment by Dray56** [comment]
Hi @codeyogi911 ,
As part of the CF V3 API spec, deploying to html5 applications repository is not supported. However you can try deploying via the MTA resource which will be available as part of the next release.

**Reply by codeyogi911** [reply]
@Dray56 thanks for the response, what would that version be?

**Reply by Dray56** [reply]
Hi @codeyogi911 ,
MTAR resource is already part of main. You can clone it and build it [locally](https://github.com/SAP/terraform-provider-cloudfoundry/blob/main/DEVELOPER.md#cloning-of-the-repository). Release to the terraform registry might take some time. I dont have an ETA at this point of time.

**Reply by Dray56** [reply]
Hi @codeyogi911,
MTA Resource is now part of release [1.0.0-rc1](https://registry.terraform.io/providers/SAP/cloudfoundry/latest/docs/resources/mta).

**Reply by codeyogi911** [reply]
@Dray56 i see that now! thanks it makes it much simpler. However what i dont understand is: How to ensure `mtar` is deployed again on each change? 
For example i did some change and want my apps to update?

**Reply by Dray56** [reply]
Terraform needs to detect that change to propose an update. For this purpose, we have brought in a new attribute called `source_code_hash`. When the contents of an MTAR changes, this value would ideally change and terraform should detect this change. You can use the terraform's inbuilt-function [filesha256](https://developer.hashicorp.com/terraform/language/functions/filesha256) function to do this. 

Note: The `source_code_hash` attribute has been added recently in the main branch to address this issue, so you will need to build it [locally](https://github.com/SAP/terraform-provider-cloudfoundry/blob/main/DEVELOPER.md#cloning-of-the-repository) for now :)
